Task#2

RichardVas opened the pull request:
@hegyhati 
Elkészültünk a második feladat megvalósításával.

RichardVas committed:
Add parseUnit method

Extended the fighter class with parseUnit, to process json files.

RichardVas committed:
Add parseUnit method

RichardVas committed:
Implement task#2

RichardVas committed:
Expected outcome of task#2

RichardVas committed:
Add shell script for testing

RichardVas committed:
Add 3 units for testing

RichardVas committed:
Add workflow to test task#2

RichardVas committed:
Create main.yml

RichardVas committed:
Merge pull request #2 from Teaching-projects/Task#2-1

Add workflow to test Task#2

RichardVas committed:
Add Task#2 description

RichardVas committed:
Delete main.yml

vizvezetek commented on Fighter.cpp:
```diff
@@ -20,3 +20,53 @@ std::ostream& operator<<(std::ostream& os, const Fighter& fi)
+	std::ifstream file;
+
+
+	std::string name = "";
+	int hp = -1;
+	int dmg = -1;
```
Miért hozzuk itt létre akkor is, ha a file mondjuk not found?

vizvezetek commented on Source.cpp:
```diff
@@ -3,43 +3,48 @@
-
 		if (can_attack) {
-			std::cout << attacker.getName() << "--->" << defender.getName() << std::endl;
+			/*std::cout << attacker.getName() << "--->" << defender.getName() << std::endl;*/
```
Ne tároljuk a "szemetet" a kódban.

vizvezetek commented on test.sh:
```diff
@@ -0,0 +1,19 @@
+	rm $FILE
+fi
+
+for i in units/* 
+do
+	for j in units/*
+	do
+		if [ $i != $j ]
+		then
+			./a.out $i $j >> $FILE
+		fi
+	done
+done
```
Ez script nem ellenőrzi a diff-et.

vizvezetek reviewed (commented):
A kis hibák javítása után rendben lesz.

RichardVas committed:
Add expected outcome of task#2

Fixed the new line problem

RichardVas committed:
Add check for differences

Added an additional check to the script, and updated the output generation part

RichardVas committed:
Fix the difference problem

RichardVas committed:
Remove garbage code from Source.cpp

RichardVas committed:
Remove garbage code from Fighter.cpp

RichardVas committed:
Remove unnecessary output file

RichardVas committed:
Remove unnecessary code from Fighter.cpp

RichardVas committed:
Fix unnecessary declarations

In case of not finding the appropriate file

RichardVas commented:
@hegyhati @vizvezetek 
A kért változtatásokat, hibák kijavítását elvégeztük.

vizvezetek reviewed (commented):
@hegyhati Én rendben gondolom.
